### ISSUE TITLE
New rule: `no-reference-like-url`

### ISSUE THREAD

**Issue by wooorm** [wooorm/remark-lint#36]
``` md
<!-- Good -->
[this](http://example.com) and [that][]

[that]: https://example.com

<!-- Bad -->
[this](that) and [that][]

[that]: https://example.com
```
